src/Workspaces/CSharp/Portable/Classification/SyntaxClassification/NameSyntaxClassifier.cs

```diff
@@ -304,6 +344,35 @@ private static ISymbol TryGetSymbol(NameSyntax name, SymbolInfo symbolInfo, Sema
             return symbolInfo.Symbol;
         }
 
+        private bool TryClassifyAsNamespace(
+            NameSyntax name,
+            SymbolInfo symbolInfo,
+            ArrayBuilder<ClassifiedSpan> result)
+        {
+            if (name is IdentifierNameSyntax
+                && symbolInfo.Symbol == null
+                && IsInNamespaceDeclarationOrUsingDirective(name))
+            {
+                result.Add(new ClassifiedSpan(name.Span, ClassificationTypeNames.NamespaceName));
+                return true;
+            }
+
+            return false;
+        }
+
+        private static bool IsInNamespaceDeclarationOrUsingDirective(NameSyntax name)
+        {
+            while (name.Parent is NameSyntax parentName)
+            {
+                name = parentName;
+            }
+
+            // Because this check runs after the TryClassifySymbol we can assume
+            // non-classified names are namespace names.
+            return name.IsParentKind(SyntaxKind.NamespaceDeclaration)
+                || name.IsParentKind(SyntaxKind.UsingDirective);
```

Comment:
what about `using x = ...SomeClass`?  How do you know that 'x' or 'SomeClass' are namespaces?

Reply:
Yeah, my comment was trying to convey that, in the case of a type alias, the type would be classified by `TryClassifySymbol` which runs first. I will find better words to express that and maybe move `TryClassifyAsNamespace` to be invoked by `TryClassifySymbol` to make the dependency more obvious.

Reply:
Sounds good. One thing i struggled with in this PR was the relation between all the different classifiers and who was responsible for what.  Having some more comments might be helpful :)